plugins/cr8-copy-s3/src/main/java/io/crate/copy/s3/S3FileInput.java

```diff
@@ -146,6 +152,38 @@ static URI toPreGlobUri(URI fileUri) {
         return preGlobUri;
     }
 
+    @VisibleForTesting
+    static URI reformat(URI uri) {
+        assert "s3".equals(uri.getScheme());
+        if (uri.getHost() != null) {
+            if (uri.getPath() == null || uri.getPort() == -1) {
+                // This addresses the following issues:
+                // ex1) s3://bucket
+                //      URI considers 'bucket' as a host name,
+                //      but CrateDB (which accepts s3:// formatting) expects 'bucket' to be part of path component and eventually as bucket name
+                // ex2) s3://bucket/key1/key2
+                //      URI parses this into host: bucket and path: /key1/key2
+                //      but CrateDB expects this to become path: /bucket/key1/key2 and eventually as bucket: bucket and key: /key1/key2
+                return URI.create("s3://"
+                                  + (uri.getRawUserInfo() == null ? "" : uri.getRawUserInfo() + "@")
+                                  + "/"
+                                  + uri.getHost()
+                                  + uri.getPath());
+            }
+        }
+        return uri;
+    }
+
+    private static URI fixSlashes(URI uri) {
```

Comment:
Could this be done as part of the initial reformat? Or why is this a separate step?